R/periods.r

```diff
@@ -418,13 +421,15 @@ parse_period <- function(x) {
 }
 
 .period_from_num <- function(num, units) {
+  if (length(num) == 0)
+    return(new("Period", numeric()))
 
   if (!is.numeric(num)) {
     stop(sprintf("First argument to `period` constructor must be character or numeric. Supplied object of class '%s'", class(num)))
   }
 
-  if (is.interval(num))
-    stop("Interval objects cannot be used as input to 'period' constructor. Plese use 'as.period'.")
+  if (class(num)[[1]] %in% c("Interval", "Duration"))
```

Comment:
This will fail for subclasses of Interval and Duration, which is probably not what you want.

Reply:
yes, this is what we use elsewhere for efficiency reasons (I guess). Let me change all of that.